main.c

```diff
@@ -0,0 +1,121 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#define MAX_LEN 100
+
+// Function to check if the character is a valid digit or operator
+int isValidCharacter(char c) {
+    return isdigit(c) || c == '+' || c == '-' || c == '*' || c == '/';
+}
+
+// Function to handle the precedence of operators
+int precedence(char operator) {
+    if (operator == '*' || operator == '/') {

```

Comment:
According to DMAS. Division should happen first. Next multiplication and so on. But here I can see the precedence is set as same for both multiplication and division.

Will it affect logically?
If yes, do necessary changes.